plugins/bookshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,111 @@
+        if util.fileExists(k) then
+            local directory, filename = util.splitFilePathName(k) -- luacheck: no unused
+            Dispatcher:registerAction(k, {category="none", event="BookShortcut", title=filename, filemanager=true, arg=k,})
+        end
+    end
+end
+
+function BookShortcuts:onBookShortcut(path)
+    if util.fileExists(path) then
+        local ReaderUI = require("apps/reader/readerui")
+        ReaderUI:showReader(path)
+    end
+end
+
+function BookShortcuts:init()
+    self:onDispatcherRegisterActions()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function BookShortcuts:onFlushSettings()
+    if self.shortcuts and self.updated then
+        self.shortcuts:flush()
+        self.updated = false
+    end
+end
+
+function BookShortcuts:addToMainMenu(menu_items)
+    menu_items.book_shortcuts = {
+        text = _("Book Shortcuts"),
+        sorting_hint = "more_tools",
+        sub_item_table_func = function()
+            return self:getSubMenuItems()
+        end,
+    }
+end
+
+function BookShortcuts:getSubMenuItems()
+    local sub_item_table = {
+        {
+            text = _("New"),
+            keep_menu_open = true,
+            callback = function(touchmenu_instance)
+                local path_chooser = PathChooser:new{
+                    select_file = true,
+                    select_directory = false,
+                    detailed_file_info = true,
+                    path = G_reader_settings:readSetting("home_dir"),
+                    onConfirm = function(file_path)
+                        self:addShortcut(file_path)
+                        touchmenu_instance.item_table = self:getSubMenuItems()
+                        touchmenu_instance.page = 1
+                        touchmenu_instance:updateItems()
+                    end
+                }
+                UIManager:show(path_chooser)
+            end,
+            separator = true,
+        }
+    }
+    for k,v in FFIUtil.orderedPairs(self.shortcuts.data) do
+        table.insert(sub_item_table, {
+            text = k,
+            hold_callback = function(touchmenu_instance)
+                UIManager:show(ConfirmBox:new{
+                    text = _("Do you want to delete this shortcut?"),
+                    ok_text = _("Yes"),
+                    cancel_text = _("No"),
+                    ok_callback = function()
+                        self:deleteShortcut(k)
+                        touchmenu_instance.item_table = self:getSubMenuItems()
+                        touchmenu_instance.page = 1
+                        touchmenu_instance:updateItems()
+                    end,
+                })
+            end,
+        })
+    end
+    return sub_item_table
+end
+
+function BookShortcuts:addShortcut(name)
+    self.shortcuts.data[name] = true
+    self.updated = true
+    self:onDispatcherRegisterActions()
+end
+
+function BookShortcuts:deleteShortcut(name)
+    self.shortcuts.data[name] = nil
+    self.updated = true
```

Comment:
No unregisterAction ?
If not, you could just check if it's still in `self.shortcuts.data[] ` in `BookShortcuts:onBookShortcut()`

Reply:
it wouldn't take affect anyway until you reload all plugins dependent on dispatcher, and if you are reloading plugins you are also reloading this one :)

Reply:
Implemented.
Profiles was properly refreshing. I switched gestures to using a `sub_item_table_func` to cause it to refresh properly.

now if a book is added or removed it will reflect immediately in the profiles/gestures.